nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,170 @@
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.nuxeo.ecm.core.api.security.SecurityConstants.SYSTEM_USERNAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.ACTION_NAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.INDEX_UPDATE_ALIAS_PARAM;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Endpoint to administer Elasticsearch
+ *
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the NXQL query.
+     *
+     * @see #performIndexing(String, String)
+     */
+    @POST
+    @Path("{repositoryName}/reindex")
+    public Response doIndexing(@PathParam("repositoryName") String repositoryName, @QueryParam("query") String query) {
+        return performIndexing(repositoryName, query);
+    }
+
+    /**
+     * Performs an ES indexing on the given document and its children for a given repository.
+     *
+     * @see #performIndexing(String, String)
+     */
+    @POST
+    @Path("{repositoryName}/{documentId}/reindex")
+    public Response doIndexingOnDocument(@PathParam("repositoryName") String repositoryName,
+            @PathParam("documentId") String documentId) {
+        String query = String.format("Select * From Document where %s = '%s' or %s = '%s'", //
+                NXQL.ECM_UUID, documentId, //
+                NXQL.ECM_PARENTID, documentId);
+
+        return performIndexing(repositoryName, query);
+    }
+
+    /**
+     * Gets the ES indexing status for the given indexing id.
+     *
+     * @param indexingId the indexing id of the launched indexing
+     */
+    @GET
+    @Path("indexing/{indexingId}/status")
+    public Response getIndexingStatus(@PathParam("indexingId") String indexingId) {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        BulkStatus bulkStatus = bulkService.getStatus(indexingId);
+        Map<String, BulkStatus.State> indexingStatus = new HashMap<>();
+        indexingStatus.put("status", bulkStatus.getState());
+
+        return Response.status(SC_OK).entity(indexingStatus).build();
```

Comment:
yes, as i explained, my idea is to hide the implementation, from my point of view the client doesn't have to know that our implementation is BAF to index. It's why i decide to use a variable name as `indexingId` instead of `commandId`, and i didn't return the BulkStatus only the id. This way all is hided from outside,

Reply:
Don't forget that users in this case are us, and by hiding an implementation detail, you also hide all metrics owned by the bulk status which are interested for the user. Last point, this is not consistent with what was done in `RenditionsObject`.

Reply:
i was thinking that a customer user will use these endpoints, like it's done on JSF UI by admin users.